client/src/pages/record.tsx

```diff
@@ -164,150 +189,183 @@ function RecordPage(props: {
       />
-      <div data-cy="progress" className={classes.block}>
-        <Typography
-          variant="h6"
-          className={classes.title}
-          style={{ textAlign: "center" }}
-        >
-          Questions {recordState.answerIdx + 1} / {recordState.answers.length}
-        </Typography>
-        <ProgressBar
-          value={recordState.answerIdx + 1}
-          total={recordState.answers.length}
-        />
-      </div>
-      {mentor.mentorType === MentorType.VIDEO ? (
-        <VideoPlayer classes={classes} recordState={recordState} />
-      ) : undefined}
-      <div data-cy="question" className={classes.block}>
-        <Typography className={classes.title}>Question:</Typography>
-        <FormControl className={classes.inputField} variant="outlined">
-          <OutlinedInput
-            data-cy="question-input"
-            multiline
-            value={curAnswer?.editedAnswer.question?.question}
-            disabled={curAnswer?.editedAnswer.question?.mentor !== mentor._id}
-            onChange={(e) => {
-              if (curAnswer?.editedAnswer.question) {
-                recordState.editAnswer({
-                  question: {
-                    ...curAnswer?.editedAnswer.question,
-                    question: e.target.value,
-                  },
-                });
-              }
-            }}
-            endAdornment={
-              <InputAdornment position="end">
-                <IconButton
-                  data-cy="undo-question-btn"
-                  disabled={
-                    curAnswer?.editedAnswer.question?.question ===
-                    curAnswer?.answer.question?.question
-                  }
-                  onClick={() =>
+
+      {displayRecordingPage ? (
+        <div>
+          <div data-cy="progress" className={classes.block}>
+            <Typography
+              variant="h6"
+              className={classes.title}
+              style={{ textAlign: "center" }}
+            >
+              Questions {recordState.answerIdx + 1} /{" "}
+              {recordState.answers.length}
+            </Typography>
+            <ProgressBar
+              value={recordState.answerIdx + 1}
+              total={recordState.answers.length}
+            />
+          </div>
+          {mentor.mentorType === MentorType.VIDEO ? (
+            <VideoPlayer classes={classes} recordState={recordState} />
+          ) : undefined}
+          <div data-cy="question" className={classes.block}>
+            <Typography className={classes.title}>Question:</Typography>
+            <FormControl className={classes.inputField} variant="outlined">
+              <OutlinedInput
+                data-cy="question-input"
+                multiline
+                value={curAnswer?.editedAnswer.question?.question}
+                disabled={
+                  curAnswer?.editedAnswer.question?.mentor !== mentor._id
+                }
+                onChange={(e) => {
```

Comment:
seems like a lot of logic inside of a component's event handler, and then (maybe?) is repeated below w some minor variation? Can this be broken out into a named function (private to record.tsx still)?